dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListener.cpp

```diff
@@ -1,12 +1,16 @@
-﻿#include "pch.h"
+﻿#pragma once
+
+#include "pch.h"
 #include "../PushNotificationUtility.h"
 
 HRESULT NotificationListener::RuntimeClassInitialize(
     std::shared_ptr<ForegroundSinkManager> foregroundSinkManager,
+    std::shared_ptr<ToastRegistrationManager> toastRegistrationManager,
     std::wstring appId,
-    std::wstring processName)
+    std::wstring processName) noexcept
```

Comment:
Is shared_ptr assignment exception safe?

Reply:
For sure wstring is not, so this function can't be marked as noexcept.

Reply:
It has to be noexcept try catch return. Methods that return HRESULT should not throw exceptions.

Reply:
Not resolved yet.